storage/tianmu/core/engine.cpp

```diff
@@ -769,17 +769,13 @@ void Engine::Rollback(THD *thd, bool all, bool force_error_message) {
 }
 
 void Engine::DeleteTable(const char *table, [[maybe_unused]] THD *thd) {
-  {
-    std::unique_lock<std::shared_mutex> index_guard(tables_keys_mutex);
-    index::TianmuTableIndex::DropIndexTable(table);
-    m_table_keys.erase(table);
-  }
   {
     std::unique_lock<std::shared_mutex> mem_guard(mem_table_mutex);
     TianmuMemTable::DropMemTable(table);
     mem_table_map.erase(table);
   }
 
+  DeleteTableIndex(table, thd);
```

Comment:
Here, we should do check whether operation is ok or not. if not, it should issue an error message.

Reply:
added return value to check